packages/atomic/src/components/result-lists/atomic-result-children/atomic-result-children.tsx

```diff
@@ -140,31 +162,94 @@ export class AtomicResultChildren {
     );
   }
 
-  public render() {
-    if (!this.ready) return null;
-    if (this.templateHasError) return <slot></slot>;
+  private shouldShowChildrenWrapper() {
     const result = this.getResult();
-    if (result.children.length) {
+    if (!this.initialChildren.length && this.hideResults) return false;
+    return (
+      Boolean(result.children.length) ||
+      result.isLoadingMoreResults ||
+      this.showAllResults
+    );
+  }
+
+  public componentWillUpdate() {
+    if (this.getResult().children && !this.initialChildren) {
+      this.initialChildren = this.getResult().children;
+    }
+    if (this.getResult().isLoadingMoreResults && !this.showAllResults) {
+      this.showAllResults = true;
+    }
+  }
+
+  private getComponents() {
+    const components = [];
+    const result = this.getResult();
+    const hasResults = Boolean(result.children.length);
+    if (hasResults) {
+      components.push(<slot name="before-children"></slot>);
+    }
+    if (result.isLoadingMoreResults) {
```

Comment:
I think this might be easier to read and reason about if it was structured into different scenarios with dedicated functions to handle them, with all the boolean check aligned in a "flat" manner.

For example:
```typescript
[... etc ...]
if(result.isLoadingMoreResult()) {
   return this.getComponentsForLoadingState()
}

if(this.hasResults) {
   return this.getComponentsForHasResultsState()
}

if(this.showAllResults) {
   return this.getComponentsForNoResultsState()
}
[... etc ...]


private get hasResults() {
  return this.getResult().children.length !== 0;
}

private getComponentsForLoadingState() {
   return [ 
           this.hasResult ? <slot name="before-children"></slot> : null, 
          <ListDisplayResultsPlaceholder
              resultsPerPageState={{
                numberOfResults: this.getResult().children.length || 2,
              }}
              density={this.displayConfig.density}
              imageSize={this.imageSize || this.displayConfig.imageSize}
              isChild
          />
          this.hasResult ? , <slot name="after-children"></slot>) : null
].filter(component=> component !== null)

}

[... etc..]
```

I think this would help to see each state as individual switch cases that needs to independently render different components, VS all intermixed.

Reply:
I agree, what I did isn't great... I'm updating!